### PR TITLE
ADJUST1-452 DR Edit remand

### PR DESCRIPTION
**Pull request by pavilionsahota**

### CODE REVIEW

File: server/model/remandChangeModel.ts [hunk 1]
```diff
@@ -18,4 +19,31 @@ export default class RemandChangeModel {
   public remandRelatedValidationSummary() {
     return remandRelatedValidationSummary(this.calculatedUnusedDeductions?.validationMessages)
   }
+
```
Comment (yasinmustafa) [inline server/model/remandChangeModel.ts:22]: ideally we try and use more functional programming techniques in modern typescript, e.g. moving away from using traditional for loops and mutable variables.. something like this. 

```
public isModified(): boolean {
  if (!this.dbAdjustment) {
    return false;
  }

  const sessionCharges = this.adjustment.remand.chargeId.sort((a, b) => a - b);
  const dbCharges = this.dbAdjustment.remand.chargeId.sort((a, b) => a - b);

  const chargeIdModified = !sessionCharges.every((chargeId, index) => chargeId === dbCharges[index]);

  const dateModified = this.adjustment.toDate !== this.dbAdjustment.toDate || 
                       this.adjustment.fromDate !== this.dbAdjustment.fromDate;

  return chargeIdModified || dateModified;
}

```

File: server/views/pages/adjustments/remand/edit.njk [hunk 2]
```diff
@@ -57,18 +57,16 @@
...
-
       {% if model.showUnusedMessage %}
         <p class="govuk-body">The updates will change the amount of unused deductions. Check the unused remand alert on NOMIS.</p>
       {% endif %}
 
       <form method="post">
         <input type="hidden" name="_csrf" value="{{ csrfToken }}"/>
         <div class="govuk-button-group">
-          {% if not model.remandRelatedValidationSummary().errorList.length %}
+          {% if not model.remandRelatedValidationSummary().errorList.length and model.isModified() %}
```
Comment (yasinmustafa) [inline server/views/pages/adjustments/remand/edit.njk:67]: is it worth adding a unit test for this?

File: server/model/remandChangeModel.ts (hunk 1 above)
Comment (pavilionsahota) [reply server/model/remandChangeModel.ts:22]: I've adjusted this as you've suggested

File: server/views/pages/adjustments/remand/edit.njk (hunk 2 above)
Comment (pavilionsahota) [reply server/views/pages/adjustments/remand/edit.njk:67]: I've written 2 unit tests for this

**Review by yasinmustafa — APPROVED** [review]
lgtm

**Pull request merged**